pyopencl/array.py

```diff
@@ -1279,10 +1279,14 @@ def _zero_fill(self, queue=None, wait_for=None):
         if not self.size:
             return
 
-        if (
-                queue._get_cl_version() >= (1, 2)
-                and cl.get_cl_header_version() >= (1, 2)):
-
+        cl_version_gtr_1_2 = (
+            queue._get_cl_version() >= (1, 2)
+            and cl.get_cl_header_version() >= (1, 2)
+        )
+        on_nvidia = queue.device.vendor.startswith("NVIDIA")
+
+        # circumvent bug with large buffers on NVIDIA (gh-395)
```

Comment:
```suggestion
        # circumvent bug with large buffers on NVIDIA
        # https://github.com/inducer/pyopencl/issues/395
```
(Clearer and directly clickable)